### PR TITLE
Migrate to .NET 6.0

### PR DESCRIPTION
**Pull request by wsugarman**
## Description
Update all projects to .NET 6 and Azure Functions to v4.

Checklist
- [x] Update *global.json*
- [x] Update target frameworks to `net6.0`
  - [x] Dicom server
  - [x] Dicom cast
  - [x] Scale testing
- [x] Update Azure Functions to v4
- [x] Update docker images
- [x] Address analyzer issues
  - [x] Use `AsSpan` over `Substring`
  - [x] Use PascalCase for template placeholders
  - [x] (Ignored) Use logging delegates for performance
  - [x] (Somewhat ignored) Ensure message templates are constant
- [x] Fix tests/breakage due to underlying changes
- [x] Update NuGet packages
- [x] Update healthcare-shared-components due to SQL tooling

## Related issues
[AB#86626](https://microsofthealth.visualstudio.com/Health/_workitems/edit/86626)
[AB#86629](https://microsofthealth.visualstudio.com/Health/_workitems/edit/86629)

## Testing
Locally and via pipeline

### CODE REVIEW

File: test/Microsoft.Health.Dicom.Tests.Integration/Features/TranscoderTests.cs [hunk 1]
```diff
@@ -152,7 +152,7 @@ private string GetFramesHashCode(DicomFile dicomFile)
 
         private string GetByteArrayHashCode(byte[] byteArray)
         {
-            return Convert.ToBase64String(new SHA1Managed().ComputeHash(byteArray));
+            return Convert.ToBase64String(HashAlgorithm.Create(nameof(SHA1)).ComputeHash(byteArray));
```
Comment (brandonpollett) [inline test/Microsoft.Health.Dicom.Tests.Integration/Features/TranscoderTests.cs:155]: We need to use something like SHA256.

**Review by brandonpollett — APPROVED** [review]

File: converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/ChangeFeedProcessor.cs [hunk 2]
```diff
@@ -81,11 +81,11 @@ public async Task ProcessAsync(TimeSpan pollIntervalDuringCatchup, CancellationT
                         if (!(changeFeedEntry.Action == ChangeFeedAction.Create && changeFeedEntry.State == ChangeFeedState.Deleted))
                         {
                             await _fhirTransactionPipeline.ProcessAsync(changeFeedEntry, cancellationToken);
-                            _logger.LogInformation("Successfully processed DICOM event with SequenceID: {sequenceId}", changeFeedEntry.Sequence);
+                            _logger.LogInformation("Succesfully processed DICOM event with SequenceID: {SequenceId}", changeFeedEntry.Sequence);
```
[the remark this answers is not in the document]
Comment (wsugarman) [reply converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/ChangeFeedProcessor.cs:84]: Whoops! Not my best merge lol

File: src/Microsoft.Health.Dicom.Api/Extensions/HttpResponseExtensions.cs [hunk 3]
```diff
@@ -15,12 +15,14 @@ internal static class HttpResponseExtensions
     {
         public const string ErroneousAttributesHeader = "erroneous-dicom-attributes";
 
+        private static readonly Uri UnusedRoot = new Uri("http://unused/", UriKind.Absolute);
```
[the remark this answers is not in the document]
Comment (wsugarman) [reply src/Microsoft.Health.Dicom.Api/Extensions/HttpResponseExtensions.cs:18]: That sounds better

File: src/Microsoft.Health.Dicom.SqlServer/Extensions/ILoggerExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Health.Dicom.SqlServer.Extensions
+{
+    internal static class ILoggerExtensions
+    {
+        public static void LogStoredProcedureSuccess(this ILogger logger, string name, Stopwatch sw)
```
[the remark this answers is not in the document]
Comment (wsugarman) [reply src/Microsoft.Health.Dicom.SqlServer/Extensions/ILoggerExtensions.cs:13]: Oh good call! I don't use these enough

File: test/Microsoft.Health.Dicom.Tests.Integration/Features/TranscoderTests.cs (hunk 1 above)
Comment (wsugarman) [reply test/Microsoft.Health.Dicom.Tests.Integration/Features/TranscoderTests.cs]: Our tests files make the assumption we use SHA1, so I'll make this change in another PR

**Pull request merged**